tests/aggregation/tests.py

```diff
@@ -2116,6 +2116,19 @@ def test_multiple_aggregate_references(self):
             },
         )
 
+    def test_group_by_reference_subquery(self):
+        qs = Publisher.objects.filter(
+            pk__in=Author.objects.annotate(
+                publisher_id=F("book__publisher"),
+            )
+            .values("publisher_id")
+            .annotate(
+                cnt=Count("*"),
+            )
+            .values("publisher_id")
```

Comment:
As far as I'm aware, the second `values()` is unnecessary to reproduce the issue
```suggestion
```